scripts/lernaVersion.sh

```diff
@@ -19,15 +19,27 @@ rootDir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && pwd )"
 
 lernaVersionArg="$1"
 if [[ -z $lernaVersionArg ]]; then
-  echo 'Please provide argument for "lerna version".'
+  echo '[ERROR] Please provide argument for "lerna version".'
+  exit 1
+fi
+
+npmVersion=$(node -e "process.stdout.write(require('./package.json').engines.npm)");
+isSatisfiedNpm=$(node -e "process.stdout.write(require('semver').satisfies('$(npm --version)', '${npmVersion}').toString())");
+
+nodeVersion=$(node -e "process.stdout.write(require('./package.json').engines.node)");
+isSatisfiedNode=$(node -e "process.stdout.write(require('semver').satisfies('$(node --version)', '${nodeVersion}').toString())");
+
+# Check node and npm version compatible with package.json
+if [[ $isSatisfiedNpm != 'true' || $isSatisfiedNode != 'true' ]]; then
+  echo '[ERROR] node version or npm version is not compatible with package.json'
```

Comment:
Optional future improvement idea: It would be nice if this distinguished between node/npm and showed the current vs supported version.

Reply:
Nice catch! done